software/controller/lib/core/interpolant.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <stdint.h>
+
+#include "nvparams.h"
+#include "vars.h"
+
+// This class abstracts a lookup table that is stored in EEPROM, accessible through the debug
+// interface, and used for linear interpolation with input between 0 and 1.
```

Comment:
Maybe explain that it clamps to those bounds? Or this could be part of some doxygen block next to function.

Reply:
sure thing